src/v/ssx/work_queue.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "ssx/work_queue.h"
+
+#include <exception>
+
+namespace ssx {
+work_queue::work_queue(error_reporter_fn fn)
+  : _error_reporter(std::move(fn)) {}
+
+void work_queue::submit(ss::noncopyable_function<ss::future<>()> fn) {
+    if (_as.abort_requested()) {
+        return;
```

Comment:
I thought about this - right now there isn't a use case because starting a wasm engine isn't interruptible at the moment and that's the main source of latency I'd assume. Do you think it's worth future proofing?

Reply:
By future proofing I mean adding now

Reply:
>Do you think it's worth future proofing?

nah